client/src/components/ProgressBar/progress-bar.tsx

```diff
@@ -0,0 +1,98 @@
+import React, { useState, useEffect } from 'react';
+import { useTranslation } from 'react-i18next';
+import BezierEasing from 'bezier-easing';
+
+interface ProgressBarProps {
+  completedPercent: number;
+  isLegacyCertfOpen?: boolean;
+}
+
+function ProgressBar({
+  completedPercent,
+  isLegacyCertfOpen
+}: ProgressBarProps): JSX.Element {
+  const [shownPercent, setShownPercent] = useState(0);
+  const [progressInterval, setProgressIntervall] = useState(0);
+
+  const { t } = useTranslation();
+  const animateProgressBar = (completedPercent: number) => {
+    const easing = BezierEasing(0.2, 0.5, 0.4, 1);
+
+    if (completedPercent > 100) completedPercent = 100;
+    if (completedPercent < 0) completedPercent = 0;
+
+    const transitionLength = completedPercent * 10 + 750;
+    const intervalLength = 10;
+    const intervalsToFinish = transitionLength / intervalLength;
+    const amountPerInterval = completedPercent / intervalsToFinish;
+    let percent = 0;
+
+    const myInterval = window.setInterval(() => {
+      percent += amountPerInterval;
+
+      if (percent > completedPercent) percent = completedPercent;
+
+      setShownPercent(
+        Math.round(completedPercent * easing(percent / completedPercent))
+      );
+      if (percent >= completedPercent) clearInterval(myInterval);
+    }, intervalLength);
+
+    setProgressIntervall(myInterval);
+  };
+  useEffect(() => {
+    let applyAnimation = true;
+    setTimeout(() => {
+      if (applyAnimation) animateProgressBar(completedPercent);
+    }, 50);
+    return () => {
+      if (progressInterval !== null) {
+        clearInterval(progressInterval);
+        applyAnimation = false;
+      }
+    };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [completedPercent]);
+
+  const defaultMargin = '32px 0px 32px 0px';
+  const defaultStyle = (progressBarWrap: boolean) => {
+    return !isLegacyCertfOpen
+      ? {
+          width: '100%',
+          margin: progressBarWrap ? defaultMargin : 'none'
+        }
+      : {};
+  };
+
+  return (
+    <div
+      className='progress-bar-wrap'
+      aria-label={t('learn.percent-complete', {
+        percent: completedPercent
+      })}
```

Comment:
```suggestion
```
The JAWS screen reader does not respect `aria-label` on the `div` element. So we'll need to add this text using sr-only text instead.